assets/javascripts/initializers/communitarian.js.es6

```diff
@@ -4,10 +4,59 @@ import { setDefaultHomepage } from "discourse/lib/utilities";
 import TopicController from "discourse/controllers/topic";
 import ResolutionController from "../controllers/resolution-controller";
 import discourseComputed from "discourse-common/utils/decorators";
+import { registerUnbound } from "discourse-common/lib/helpers";
 import { ajax } from "discourse/lib/ajax";
 import { extractError } from "discourse/lib/ajax-error";
 
 function initializeCommunitarian(api) {
+  api.modifyClass("controller:discovery/categories", {
+    isHomePage: isHomePageField.isHomePage,
+
+    @discourseComputed(
+      "model.categories",
+      "siteSettings.landing_categories_length"
+    )
+    limitedCategories(categories, length) {
+      this.set(
+        "model.categories.content",
+        categories.content.slice(0, length || 4)
+      );
+      return categories;
+    },
+  });
+
+  api.modifyClass("controller:navigation/categories", isHomePageField);
+
+  registerUnbound('compare', function(v1, operator, v2) {
+    let operators = {
+      '==': (l, r) => l == r,
+      '!=': (l, r) => l != r,
+      '>':  (l, r) => l >  r,
+      '>=': (l, r) => l >= r,
+      '<':  (l, r) => l <  r,
+      '<=': (l, r) => l <= r,
+    };
+    return operators[operator] && operators[operator](v1, v2);
+  });
+
+  api.modifyClass("controller:discovery/categories", {
```

Comment:
All `controller:discovery/categories` modifications can be completely removed from here (both places)